polkadot/node/subsystem-bench/src/subsystem-bench.rs

```diff
@@ -198,6 +216,52 @@ impl BenchCli {
 	}
 }
 
+#[cfg(target_os = "linux")]
+fn is_valgrind_mode() -> bool {
+	!matches!(crabgrind::run_mode(), crabgrind::RunMode::Native)
+}
+
+#[cfg(not(target_os = "linux"))]
+fn is_valgrind_mode() -> bool {
+	false
+}
+
+/// Start collecting cache misses data
+#[cfg(target_os = "linux")]
+fn valgrind_start() {
+	crabgrind::cachegrind::start_instrumentation();
+}
+
+#[cfg(not(target_os = "linux"))]
+fn valgrind_start() {}
+
+/// Stop collecting cache misses data
+#[cfg(target_os = "linux")]
+fn valgrind_stop() {
+	crabgrind::cachegrind::stop_instrumentation();
+}
+
+#[cfg(not(target_os = "linux"))]
+fn valgrind_stop() {}
+
+#[cfg(target_os = "linux")]
+fn valgrind_init() -> eyre::Result<()> {
+	use std::os::unix::process::CommandExt;
+	std::process::Command::new("valgrind")
```

Comment:
it doesn't look that we get an error printed if `valgrind` is missing

Reply:
Good catch!